### ISSUE TITLE
Issue in index iterator on cleanup branch 

### ISSUE THREAD

**Issue by gshipman** [laristra/flecsi#178]
FleCSI: 78b7d050ff9fe6555c8bc808337693c1f270d9aa
Legion: StanfordLegion/legion@4d0c87a85f4c295d8be85f9b9d38395e0375ca64
Legion cmake is: 
cmake .. -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=/home/gshipman/local  -DCMAKE_PREFIX_PATH=/home/gshipman/local/ -DGASNet_CONDUIT=mpi  -DLegion_USE_GASNet=ON -DGASNet_INCLUDE_DIR=/home/gshipman/local/include -DCMAKE_CXX_FLAGS="-DPRIVILEGE_CHECKS -DDEBUG_REALM -DDEBUG_LEGION -DBOUNDS_CHECKS -ggdb" 

FleCSI cmake is: 
```
cmake ..  -DENABLE_MPI=ON -DMPI_LIBRARY=/home/gshipman/local/lib/libmpi.so -DENABLE_UNIT_TESTS=ON -DFLECSI_RUNTIME_MODEL=mpilegion -DCereal_INCLUDE_DIR=/home/gshipman/local/include/cereal -DLegion_ROOT=/home/gshipman/local  -DENABLE_PARTITION=ON  -DCMAKE_CXX_FLAGS="-DPRIVILEGE_CHECKS -DDEBUG_REALM -DDEBUG_LEGION -DBOUNDS_CHECKS -ggdb" 
```

When running sprint: 
```
/flecsi-dd/build/test/execution$ mpirun  -n 2 ./sprint 
```
I get an assertion failure on an index iterator                                                                                                                                                                     

```
(gdb) bt 
#0  0x00007f3bded64ae9 in raise (sig=6) at ../sysdeps/unix/sysv/linux/pt-raise.c:35
#1  <signal handler called>
#2  0x00007f3bde11ce55 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#3  0x00007f3bde11ea5a in __GI_abort () at abort.c:89
#4  0x00007f3bde1156f7 in __assert_fail_base (fmt=<optimized out>, assertion=assertion@entry=0x50d1c9 "!finished", file=file@entry=0x50d178 "/home/gshipman/local/include/legion/legion.inl", line=line@entry=1480, 
    function=function@entry=0x50d800 <Legion::IndexIterator::next()::__PRETTY_FUNCTION__> "ptr_t Legion::IndexIterator::next()") at assert.c:92
#5  0x00007f3bde1157a2 in __GI___assert_fail (assertion=0x50d1c9 "!finished", file=0x50d178 "/home/gshipman/local/include/legion/legion.inl", line=1480, 
    function=0x50d800 <Legion::IndexIterator::next()::__PRETTY_FUNCTION__> "ptr_t Legion::IndexIterator::next()") at assert.c:101
#6  0x00000000004bbc7c in Legion::IndexIterator::next (this=0x7f3b57dfd920) at /home/gshipman/local/include/legion/legion.inl:1480
#7  0x00000000004f909f in flecsi::execution::legion_dpd::init_connectivity_task (task=0x7f3b58008f00, regions=std::vector of length 5, capacity 8 = {...}, ctx=0x7f3b5804cf20, runtime=0x7f3b60004600)
    at /home/gshipman/local/src/flecsi-dd/flecsi/execution/legion/dpd.cc:129
#8  0x0000000000509468 in Legion::LegionTaskWrapper::legion_task_wrapper<&flecsi::execution::legion_dpd::init_connectivity_task> (args=0x7f3b58042880, arglen=8, userdata=0x0, userlen=0, p=...)
    at /home/gshipman/local/include/legion/legion.inl:1879
#9  0x00007f3bdf743ce2 in Realm::LocalTaskProcessor::execute_task (this=0x23d3510, func_id=25, task_args=...) at /home/gshipman/local/src/legion/runtime/realm/proc_impl.cc:912
#10 0x00007f3bdf786222 in Realm::Task::execute_on_processor (this=0x7f3b5804e8e0, p=...) at /home/gshipman/local/src/legion/runtime/realm/tasks.cc:167
#11 0x00007f3bdf788fc2 in Realm::UserThreadTaskScheduler::execute_task (this=0x23d3880, task=0x7f3b5804e8e0) at /home/gshipman/local/src/legion/runtime/realm/tasks.cc:1059
#12 0x00007f3bdf787539 in Realm::ThreadedTaskScheduler::scheduler_loop (this=0x23d3880) at /home/gshipman/local/src/legion/runtime/realm/tasks.cc:591
#13 0x00007f3bdf78d018 in Realm::Thread::thread_entry_wrapper<Realm::ThreadedTaskScheduler, &Realm::ThreadedTaskScheduler::scheduler_loop> (obj=0x23d3880)
    at /home/gshipman/local/src/legion/runtime/realm/threads.inl:131
#14 0x00007f3bdf7995c0 in Realm::UserThread::uthread_entry () at /home/gshipman/local/src/legion/runtime/realm/threads.cc:895
#15 0x00007f3bde1312c0 in ?? () from /lib64/libc.so.6
#16 0x0000000000000000 in ?? ()
```

**Comment by gshipman** [comment]
Possible fix, but might cause other issues: 128115e87f2e657f985307a67ba83179b8510713

**Comment by nickm319** [comment]
Disabled connectivity, investigating.
